Improve learn pages for configurable variables (#2087)
ballerina-platform/ballerina-dev-website: learn/user-guide/running-ballerina-code.md

- [Running a Package](#running-a-package)
- [Configuring Your Ballerina Runtimes](#configuring-your-ballerina-runtimes)
  - [Ballerina Runtime Configurable Variables](#ballerina-runtime-configurable-variables)

>   - [Configuring Sensitive Data as configurable variables](#configuring-sensitive-data-as-configurable-variables)

Reviewer: ```suggestion
  - [Configuring Sensitive Data as Configurable Variables](#configuring-sensitive-data-as-configurable-variables)
```

### Ballerina Runtime Configurable Variables

A Ballerina runtime can be configured using configurable variables.

> The values for `configurable` variables can be provided through CLI parameters, and configuration files, with 

Reviewer: ```suggestion
The values for `configurable` variables can be provided through CLI parameters and configuration files with 
```

> loading and override precedence in the same order.

Reviewer 2: Shall we break this into two sentences? Lats part we can say 
When loading the values to configurable variables, command-line args get the higher precedence. Then configuration TOML files based on how the files are located.

> You may or may not provide configuration values for configurable variables that are used in the program.

Reviewer 2: Do we need this?

> User may or may not provide configuration values for configurable variables that are used in the program.

Reviewer: ```suggestion
You may or may not provide configuration values for configurable variables that are used in the program.
```

}
```

> `?` denotes that `id` is a required configuration, hence the configuration must specify a value for key `id`.

Reviewer: ```suggestion
`?` denotes that `id` is a required configuration. Hence, the configuration must specify a value for the `id` key.
```

> If a default value assigned, the configuration is optional, hence the configuration may or may not contain a values for

Reviewer: ```suggestion
If a default value is assigned, the configuration is optional. Hence, the configuration may or may not contain values for the
```

> configurable variables `name` and `married`.

Reviewer: ```suggestion
`name` and `married` configurable variables.
```

Consider the below `Config.toml` file.
```toml
id = 1001
name = "Jhone"
```

> Since `Config.toml` file contains a value for key `name`, the program default

Reviewer: ```suggestion
Since the `Config.toml` file contains a value for the `name` key, the program default
```

Married : true
```

When running a program with configurable values, Ballerina locates the TOML files in the following ways:

- From an environment variable with the name `BAL_CONFIG_FILES` that provides a list of paths to the TOML files
  separated by the OS-specific separator. The file precedence order will be as 
  specified in the environment variable.
  
- From an environment variable with the name `BAL_CONFIG_DATA` that contains the content of the configuration TOML 
  file.
  
- If the above environment variables are not specified, the configuration file is located in the current directory
  with the file name `Config.toml` by default.

Currently, TOML-based configuration is supported for configurable variables of types `int`, `float`, `boolean`, 
`string`, `decimal`, the arrays of the respective types, and table.

In the example, you can set the path to the `Config.toml` file using the following command.

```bash
$ export BAL_CONFIG_FILES = <path>
$ bal run main.bal
User ID : 1001
User Name : Jhone
Married : true
```

> It is possible to provide the values for configurable variables through CLI arguments in the format of `-Ckey=value`.

Reviewer 2: ```suggestion
It is possible to provide the values for configurable variables through command-line arguments in the format of `-Ckey=value`.
```

> The key of a CLI argument can be specified as,

Reviewer 2: ```suggestion
The key of a command-line argument can be specified as,
```

```
key:= [[org-name .] module-name .] variable
```
The org-name and module-name is optional for the variable defined in the root module or in a single Ballerina file. 
Currently, CLI-based configuration is supported for configurable variables of types `int`, `float`, `boolean`,
`string`, `decimal`, and `xml`.

> In the example, you can use the following command to pass values from the CLI.

Reviewer 2: ```suggestion
In the example, you can use the following command to pass values from the command-line.
```

Married : true
```

When running a program with configurable values, Ballerina locates the TOML files in the following ways:

- From an environment variable with the name `BAL_CONFIG_FILES` that provides a list of paths to TOML files 
  separated by the OS-specific separator. The file precedence order will be as 
  specified in the environment variable.
  
- From an environment variable with the name `BAL_CONFIG_DATA` that contains the content of configuration TOML 
  file.
  
- If the above environment variables are not specified, the configuration file is located in the current directory 
  with the file name `Config.toml`, by default.

Currently, TOML-based configuration is supported for configurable variables of types `int`, `float`, `boolean`, 
`string`, `decimal`, the arrays of the respective types, and table.

> In the example, we can set the path to `Config.toml` file using the following command.

Reviewer: ```suggestion
In the example, you can set the path to the `Config.toml` file using the following command.
```
